packages/extension-ui/src/Popup/Signing/Request/SignArea.tsx

```diff
@@ -37,15 +37,15 @@ function SignArea({ buttonText, className, error, isExternal, isFirst, isLast, s
 
     !isExternal &&
       isSignLocked(signId)
-        .then(({ isLocked, remainingTime }) => {
-          setIsLocked(isLocked);
+        .then(({ remainingTime }) => {
+          setIsLocked(remainingTime <= 0);
           timeout = setTimeout(() => {
             setIsLocked(true);
           }, remainingTime);
 
           // if the account was unlocked check the remember me
           // automatically to prolong the unlock period
-          !isLocked && setSavePass(true);
+          remainingTime > 0 && setSavePass(true);
```

Comment:
I'd switch to the `if` statement - it's more descriptive and does not really take more code. Let's leave short-circuiting expressions for cases in which `if` is not allowed.